File: viet_nam_trip/app/Models/hoa_don.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class hoa_don extends Model
{
    use HasFactory;
    use SoftDeletes;

    public $timestamps = true;

    protected $fillable = [
        'phieu_dat_id',
        'ngay_thanh_toan',
        'tong_tien',
        'loai_thanh_toan',
        'trang_thai',
    ];

    public function phieu_dat(){
        return $this->hasOne('App\Models\phieu_dat', 'phieu_dat_id', 'id');
    }
}

File: dulich/lib/Models/hoa_don.dart
class HoaDon {
  final int id;
  final int tong_tien;
  final String loai_thanh_toan;

  HoaDon(this.id, this.tong_tien, this.loai_thanh_toan);

  HoaDon.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        tong_tien = json['tong_tien'],
        loai_thanh_toan = json['loai_thanh_toan'];
}
